src/Avalonia.Controls/TextBox.cs

```diff
@@ -217,7 +217,11 @@ public int CaretIndex
             set
             {
                 value = CoerceCaretIndex(value);
-                SetAndRaise(CaretIndexProperty, ref _caretIndex, value);
+                var changed = SetAndRaise(CaretIndexProperty, ref _caretIndex, value);
+
+                if (!changed)
```

Comment:
This looks wrong. You simply don't update when the caret index changes and if it doesn't change you update.

Reply:
Hmm, after investigating the issue, the delay happened if  `SelectionStart ` and `SelectionEnd` changed, and `CaretIndex` don't. Thus I only calls `_presenter?.InvalidateVisual()` if caret index doesn't change.

Reply:
So, I think there's no point to `InvalidateVisual` here since this will be handled internally by `TextPresenter` (if caret index changes).